memcpy_afu_ctx.c

```diff
@@ -351,13 +353,21 @@ int test_afu_memcpy(char *src, char *dst, size_t size, int count,
 
 		/* We have to do this even for the interrupt driven case because we need
 		 * to wait for this flag before setting the completion bit. */
-		for (j = 100000000;
-		     j > 0 && !(queued_we->status & MEMCPY_WE_STAT_COMPLETE); j--) {
-			/* Empty body */
-		}
+		gettimeofday(&timeout, NULL); /* reuse timeout */
+		temp.tv_sec = args->completion_timeout;
+		temp.tv_usec = 0;
+		timeradd(&timeout, &temp, &timeout);
+		for (;; gettimeofday(&temp, NULL)) {
+			if (timercmp(&temp, &timeout, >)) {
+				printf("# Timeout polling for completion\n");
+				break;
+			}
 
-		if (j == 0)
-			printf("# Timeout polling for completion\n");
+			if (queued_we->status & MEMCPY_WE_STAT_COMPLETE) {
```

Comment:
Condition can be restated as 
if (queued_we->status) { ....
so that we dont loop if AFU has reported an error.